rules/typescript.js

```diff
@@ -5,9 +5,28 @@ module.exports = {
     '@typescript-eslint/adjacent-overload-signatures': ['error'],
     '@typescript-eslint/array-type': ['error', { default: 'array-simple' }],
     '@typescript-eslint/ban-types': ['error'],
-    '@typescript-eslint/ban-ts-ignore': ['off'],
-    '@typescript-eslint/camelcase': ['error'],
-    '@typescript-eslint/class-name-casing': ['error'],
+    '@typescript-eslint/ban-ts-comment': ['off'],
+    '@typescript-eslint/naming-convention': [
+      'error',
+      {
+        selector: 'variable',
+        format: ['camelCase', 'PascalCase', 'UPPER_CASE'],
+        leadingUnderscore: 'allow',
+      },
+      {
+        selector: 'typeParameter',
+        format: ['PascalCase'],
+        prefix: ['T', 'K'],
+      },
+      {
+        selector: 'interface',
+        format: ['PascalCase'],
+        custom: {
+          regex: '^I[A-Z]',
```

Comment:
If I understand this, it forces an interface to start with `I`. We haven't used this pattern before and it's going to collide with our codebases.

Reply:
It's the other way around, it forces an interface not to start with `I` (see the `match: false` on the next line).

https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/naming-convention.md#enforce-that-interface-names-do-not-begin-with-an-i

It replaces the `@typescript-eslint/interface-name-prefix` rule that was removed.